### PR TITLE
SparkR kernel

### LINKED ISSUE #114: R support

### PR DESCRIPTION
**Pull request by jeffersonezra**
1. Add SparkR kernel.
2. Add tests
3. Support for R %sql
4. Update the for loop variable name as R doesn't allow variables to start with _
5. R's json package jsonlite returns the json result as a single string instead of a json representation (array of rows). So updated jupyter to handle this case.

Fixes #114

### CODE REVIEW

File: sparkmagic/sparkmagic/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-__version__ = '0.8.5'
+__version__ = '0.8.6'
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/__init__.py]: I think we should bump to 0.9.0.

You will also need to bump autoviz and hdijupyterutils

File: sparkmagic/sparkmagic/livyclientlib/sqlquery.py [hunk 2]
```diff
@@ -118,7 +121,16 @@ def _scala_command(self, sql_context_variable_name):
         return Command(u'{}.foreach(println)'.format(command))
 
     def _r_command(self):
-        raise NotImplementedError()
+        command = u'sql("{}")'.format(self.query)
+        if self.samplemethod == u'sample':
```
Comment (aggFTW) [inline sparkmagic/sparkmagic/livyclientlib/sqlquery.py]: @felixcheung could you please help review this part? 🙇

**Comment by aggFTW** [comment]
Just please make sure SQL queries work with test cases:

- unicode values
- empty tables
- one row
- several rows
- etc

File: sparkmagic/sparkmagic/livyclientlib/sqlquery.py [hunk 3]
```diff
@@ -59,25 +59,28 @@ def execute(self, session):
...
         if records_text == '':
             strings = []
         else:
             strings = records_text.split('\n')
         try:
             data_array = [json.JSONDecoder(object_pairs_hook=OrderedDict).decode(s) for s in strings]
-            
+
+            if kind == constants.SESSION_KIND_SPARKR and len(data_array) > 0:
+                data_array = data_array[0]
```
Comment (felixcheung) [inline sparkmagic/sparkmagic/livyclientlib/sqlquery.py]: suggest adding a newline for clarity, since this falls through to run the next `if` on L84

**Comment by aggFTW** [comment]
Fixes #114

File: sparkmagic/sparkmagic/livyclientlib/sqlquery.py [hunk 4]
```diff
@@ -118,7 +121,16 @@ def _scala_command(self, sql_context_variable_name):
...
+        command = u'sql("{}")'.format(self.query)
+        if self.samplemethod == u'sample':
+            command = u'sample({}, FALSE, {})'.format(command, self.samplefraction)
+        if self.maxrows >= 0:
+            command = u'take({},{})'.format(command, self.maxrows)
+        else:
+            command = u'collect({})'.format(command)
+        command = u'jsonlite:::toJSON({})'.format(command)
+        command = u'for ({} in ({})) {{cat({})}}'.format(constants.LONG_RANDOM_VARIABLE_NAME, command, constants.LONG_RANDOM_VARIABLE_NAME)
+        return Command(command)
```
Comment (felixcheung) [inline sparkmagic/sparkmagic/livyclientlib/sqlquery.py]: LGTM here.

**Comment by tc0312** [comment]
LGTM.

**Comment by linchan-ms** [comment]
LGTM

File: sparkmagic/sparkmagic/livyclientlib/sqlquery.py (hunk 3 above)
Comment (jeffersonezra) [inline sparkmagic/sparkmagic/livyclientlib/sqlquery.py]: Added newline

**Pull request merged**